fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/AvailableFields.java

```diff
@@ -102,14 +102,18 @@ public static AvailableFields fromIndex(@Nonnull RecordType recordType,
 
         final List<KeyExpression> keyFields = new ArrayList<>();
         final List<KeyExpression> valueFields = new ArrayList<>();
+        int offset = 0;
         if (indexTypes.getTextTypes().contains(index.getType())) {
             // Full text index entries have all of their fields except the tokenized one.
             keyFields.addAll(TextScanPlanner.getOtherFields(rootExpression));
         } else if (indexTypes.getValueTypes().contains(index.getType()) ||
-                   indexTypes.getRankTypes().contains(index.getType()) ||
-                   indexTypes.getLuceneTypes().contains(index.getType())) {
+                   indexTypes.getRankTypes().contains(index.getType())) {
             keyFields.addAll(KeyExpression.getKeyFields(rootExpression));
             valueFields.addAll(KeyExpression.getValueFields(rootExpression));
+        } else if (indexTypes.getLuceneTypes().contains(index.getType())) {
+            List<KeyExpression> nonPrimaries = rootExpression.normalizeKeyForPositions();
```

Comment:
There should be a comment here as there is for full text, that Lucene indexes also lack their non-stored fields. In the future, this will need to not assume only primary key fields are stored. And probably not leave holes where they were in the `IndexEntry`. But this is consistent for now.

Reply:
actually the primary fields are stored but as a byte ref. Thats how the doc_id is added in the tests because its not part of the root key expression, the issue isn't that we can't retrieve them but that they aren't indexed and searchable.